### ISSUE TITLE
engine: merge into the cli repository

### ISSUE THREAD

**Issue by bassosimone** [ooni/probe#1335]
We are going to use Go 1.16's new feature that enables embedding assets. To this end, it would greatly simplify our life to have a single Go repository containing all the sources. This issue is about merging the `cli` and the `engine` repositories.

We could merge the `cli` into the `engine`, or the `engine` into the `cli`. Since the `cli` is tagging publicly available version numbers we will merge the `engine` into the `cli`.

Because of Go strict semantic versioning rules, we cannot export anything from the `engine` into the new repository, otherwise, we will need to break the import path at basically every new release of `ooniprobe`.

Therefore, we will basically vendor the engine codebase inside `internal/engine`. We still to hammer out a bunch of details including what to do of the existing `engine` repository, etc.

**Comment by bassosimone** [comment]
The first problem I identified is the way in which `probe-cli` uses semantic versioning. Consider this:

```
$ pwd
/tmp
$ go version
go version go1.15.6 linux/amd64
$ 
```

**Comment by bassosimone** [comment]
Now, the second step (https://github.com/ooni/probe-cli/pull/201) is that of putting the `probe-engine` codebase into the `probe-cli` codebase. At this stage, we only care about being able to produce a working `ooniprobe` binary. Once this piece of work is done, there is extra work to ensure we can do with the `probe-cli`'s repository everything we can do with `probe-engine`.

**Comment by bassosimone** [comment]
Now, my task is to figure out which tests are failing and why. Exciting times 😬.

**Comment by bassosimone** [comment]
In https://github.com/ooni/probe-cli/pull/203 we're adding support for building miniooni from probe-cli.

**Comment by bassosimone** [comment]
In https://github.com/ooni/probe-cli/pull/204 I'm migrating probe-engine workflows to probe-cli.

**Comment by bassosimone** [comment]
The next step is automatically publishing an Android package. We're working on this in https://github.com/ooni/probe-cli/pull/205.

**Comment by bassosimone** [comment]
More refactoring underway in https://github.com/ooni/probe-cli/pull/206. It's a set of minor changes and movements after we merged the two codebases. This is ~easy and quite useful to help me identify optional complexity to kill.

**Comment by bassosimone** [comment]
With https://github.com/ooni/probe-cli/pull/211, most of the work is done. What remains to do:

- [ ] make sure the documentation makes sense and is up to date
- [ ] skip failing tests and open issues about them
- [ ] properly mark probe-engine as _clearly_ deprecated

**Comment by bassosimone** [comment]
The merge is complete. We can now continue working on the engine inside of `probe-cli`. We have clearly deprecated the `probe-engine` repository. We will keep using it as a container for past issues. We will route future contributions to `probe-cli/internal/engine`.
